src/agent/communicator/include/communicator.hpp

```diff
@@ -21,8 +20,8 @@ namespace communicator
 
         boost::asio::awaitable<void> GetCommandsFromManager();
         boost::asio::awaitable<void> WaitForTokenExpirationAndAuthenticate();
-
-        const std::string& GetToken() const;
+        boost::asio::awaitable<void> StatefulMessageProcessingTask(std::queue<std::string>& messageQueue);
+        boost::asio::awaitable<void> StatelessMessageProcessingTask(std::queue<std::string>& messageQueue);
 
     private:
         std::mutex m_exitMtx;
```

Comment:
Let's organize this better.

Reply:
About `m_exitFlag`, I think it can be initialized here and not in the constructor like `isReAuthenticating`.

Reply:
done 4199fbed2b4d220fcd998b7d0b10afd7385f5fee